server/database.go

```diff
@@ -715,3 +775,25 @@ func (s *Server) registerDatabaseRoutes(g *echo.Group) {
 		return nil
 	})
 }
+
+func (s *Server) hasRelatedOpenStatusIssue(ctx context.Context, projectID int, databaseID int) (bool, error) {
+	issueFind := &api.IssueFind{
+		ProjectID:  &projectID,
+		StatusList: []api.IssueStatus{api.IssueOpen},
+	}
+	issueList, err := s.store.FindIssue(ctx, issueFind)
```

Comment:
This could cause performance issue as issue is huge, and we cannot load them all. Let's use a SQL query to figure it out?

On the other hand, we should allow users to archive a database regardless its association with issues. We should allow archive if the database is NOT FOUND.